mne/time_frequency/psd.py

```diff
@@ -112,6 +114,202 @@ def _check_nfft(n, n_fft, n_overlap):
+    logger.info("Effective window size : %0.3f (s)" % win_size)
+    freqs = np.arange(n_fft // 2 + 1, dtype=float) * (sfreq / n_fft)
+    freq_mask = (freqs >= fmin) & (freqs <= fmax)
+    freqs = freqs[freq_mask]
+
+    # Parallelize across first N-1 dimensions
+    parallel, my_pwelch, n_jobs = parallel_func(_pwelch, n_jobs=n_jobs,
+                                                verbose=verbose)
+    x_splits = np.array_split(x, n_jobs)
+    f_psd = parallel(my_pwelch(d, noverlap=n_overlap, nfft=n_fft,
+                     fs=sfreq, freq_mask=freq_mask,
+                     welch_fun=welch)
+                     for d in x_splits)
+
+    # Combining/reshaping to original data shape
+    psds = np.concatenate(f_psd, axis=0)
+    psds = psds.reshape(np.hstack([dshape, -1]))
+    return psds, freqs
+
+
+@verbose
+def psd_welch(inst, fmin=0, fmax=np.inf, tmin=None, tmax=None, n_fft=256,
+              n_overlap=0, picks=None, proj=False, n_jobs=1, verbose=None):
+    """Compute the power spectral density (PSD) using Welch's method.
+
+    Calculates periodigrams for a sliding window over the
+    time dimension, then averages them together for each channel/epoch.
+
+    Parameters
+    ----------
+    inst : instance of Epochs or Raw or Evoked
+        The data for PSD calculation
+    fmin : float
+        Min frequency of interest
+    fmax : float
+        Max frequency of interest
+    tmin : float | None
+        Min time of interest
+    tmax : float | None
+        Max time of interest
+    n_fft : int
+        The length of the tapers ie. the windows. The smaller
+        it is the smoother are the PSDs. The default value is 256.
+        If ``n_fft > len(inst.times)``, it will be adjusted down to
+        ``len(inst.times)``.
+    n_overlap : int
+        The number of points of overlap between blocks. Will be adjusted
+        to be <= n_fft.
+        bandwidth : float
+        The bandwidth of the multi taper windowing function in Hz.
+    picks : array-like of int | None
+        The selection of channels to include in the computation.
+        If None, take all channels.
+    proj : bool
+        Apply SSP projection vectors. If inst is ndarray this is not used.
+    n_jobs : int
+        Number of CPUs to use in the computation.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    psds : ndarray, shape ([n_epochs], n_channels, n_freqs)
+        The power spectral densities. If Raw is provided,
+        then psds will be 2-D.
+    freqs : ndarray (n_freqs)
+        The frequencies.
+
+    See Also
+    --------
+    mne.io.Raw.plot_psd, mne.Epochs.plot_psd, psd_multitaper
+    """
+    # Prep data
+    data, sfreq = _check_psd_data(inst, tmin, tmax, picks, proj)
+    return _psd_welch(data, sfreq, fmin=fmin, fmax=fmax, n_fft=n_fft,
+                      n_overlap=n_overlap, n_jobs=n_jobs,
+                      verbose=verbose)
```

Comment:
Here you actually don't need to pass the `verbose` option along (I'd remove it entirely) because this function already has a `verbose` decorator. Thus whatever the input value of `verbose` was to this function will already have modified the `logging` module by the time it gets to this call in the code, and passing the `verbose` argument along to the next function will cause an unnecessary/redundant re-application of the `verbose` level. Make sense?

If not, just trust me :)